config/crd/bases/exp.infrastructure.cluster.x-k8s.io_azuremachinepools.yaml

```diff
@@ -312,9 +312,13 @@ spec:
                       should use a Spot VM
                     properties:
                       maxPrice:
+                        anyOf:
+                        - type: integer
+                        - type: string
                         description: MaxPrice defines the maximum price the user is
                           willing to pay for Spot VM instances
-                        type: number
+                        pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
```

Comment:
Sorry I'm late to the party, but, I wonder if there's any way we can change this so that it only accepts numbers and doesn't allow the textual suffixes?

Reply:
@axbarsan any reason you chose to allow textual suffixes?

Reply:
The validation was generated automatically after I changed the type to `*resource.Quantity`. I believe the reason for alowing textual suffixes is to also allow writing numbers in the "quantity" way, for not losing precision (e.g. `1.5` => `1500m`)

More info here:
https://github.com/kubernetes/apimachinery/blob/c93b0f84892eb6bcbc80b312ae70729d8168bc7e/pkg/api/resource/quantity.go#L69